tests/test_circulation_data.py

```diff
@@ -0,0 +1,265 @@
+        )
+
+        circulation_data_copy = deepcopy(circulation_data)
+
+        # If deepcopy didn't throw an exception we're ok.
+        assert circulation_data_copy is not None
+
+
+    def test_links_filtered(self):
+        # Tests that passed-in links filter down to only the relevant ones.
+        link1 = LinkData(Hyperlink.OPEN_ACCESS_DOWNLOAD, "example.epub")
+        link2 = LinkData(rel=Hyperlink.IMAGE, href="http://example.com/")
+        link3 = LinkData(rel=Hyperlink.DESCRIPTION, content="foo")
+        link4 = LinkData(
+            rel=Hyperlink.THUMBNAIL_IMAGE, href="http://thumbnail.com/",
+            media_type=Representation.JPEG_MEDIA_TYPE,
+        )
+        link5 = LinkData(
+            rel=Hyperlink.IMAGE, href="http://example.com/", thumbnail=link4,
+            media_type=Representation.JPEG_MEDIA_TYPE,
+        )
+        links = [link1, link2, link3, link4, link5]
+
+        identifier = IdentifierData(Identifier.GUTENBERG_ID, "1")
+        circulation_data = CirculationData(
+            DataSource.GUTENBERG,
+            primary_identifier=identifier,
+            links=links,
+        )
+
+        filtered_links = sorted(circulation_data.links, key=lambda x:x.rel)
+
+        eq_([link1], filtered_links)
+
+
+    def test_open_access_content_mirrored(self):
+        # Make sure that open access material links are translated to our S3 buckets, and that 
+        # commercial material links are left as is.
+        # Note: Mirroring tests passing does not guarantee that all code now 
+        # correctly calls on CirculationData, as well as Metadata.  This is a risk.
+
+        mirror = DummyS3Uploader()
+        # Here's a book.
+        edition, pool = self._edition(with_license_pool=True)
+
+        # Here's a link to the content of the book, which will be mirrored.
+        link_mirrored = LinkData(
+            rel=Hyperlink.OPEN_ACCESS_DOWNLOAD, href="http://example.com/",
+            media_type=Representation.EPUB_MEDIA_TYPE,
+            content="i am a tiny book"
+        )
+
+        # This link will not be mirrored.
+        link_unmirrored = LinkData(
+            rel=Hyperlink.DRM_ENCRYPTED_DOWNLOAD, href="http://example.com/2",
+            media_type=Representation.EPUB_MEDIA_TYPE,
+            content="i am a pricy book"
+        )
+
+        # Apply the metadata.
+        policy = ReplacementPolicy(mirror=mirror)
+        circulation_data = CirculationData(
+        	data_source=edition.data_source, 
+        	primary_identifier=edition.primary_identifier,
+        	links=[link_mirrored, link_unmirrored],
+        )
+        circulation_data.apply(pool, replace=policy)
+        
+
+        # Only the open-access link has been 'mirrored'.
+        # TODO: make sure the refactor is done right, and metadata does not upload
+        #eq_(0, len(mirror.uploaded))
+        [book] = mirror.uploaded
+
+        # TODO: make sure the refactor is done right, and circulation does upload 
```

Comment:
actually we don't need this, it's already handled by this test, but the first one still needs to be addressed